### DISCUSSION TITLE
Build WASM

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#1963] [Q&A] [answered]
i build wasm, i use:  'axmol build -p wasm', it build done but when open file.html in console:
Am I missing any steps?
im use: axmol 2.1.3 (i try windows and mac)

![image](https://github.com/axmolengine/axmol/assets/125545428/1aa3787e-28e3-44ee-801f-50859acf04ee)

**Comment by boris-741** [comment, marked as answer]
try "axmol run -p wasm" command or try start http server you self
https://github.com/axmolengine/axmol/discussions/1898#discussioncomment-9450520

**Reply by NgVThangBz** [reply]
yes, i try "axmol run -p wasm" , it work for me, thank you
Do you know the cause?

**Reply by smilediver** [reply]
Search about `SharedArrayBuffer`. IIRC it's disabled by default due to security reasons and you either need to start the browser with special flags, or you need a web server that sends special headers.

**Reply by paulocoutinhox** [reply]
Browser enable SharedArrayBuffer only locally and in HTTPS connections. You will need publish your game in a service that give you HTTPS connection (maybe all services offer it, like Cloudflare Pages, Netlify etc).

**Reply by boris-741** [reply]
> Browser enable SharedArrayBuffer only locally and in HTTPS connections. You will need publish your game in a service that give you HTTPS connection (maybe all services offer it, like Cloudflare Pages, Netlify etc).

this is not entirely true. I gave a link in the answer - it leads to an example of an http server. you need to enable COOP and COEP (you can do this on the http server)
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy

**Reply by paulocoutinhox** [reply]
Yes, because this i use my python server locally: https://github.com/paulocoutinhox/axmol-native-helper/blob/main/server.py

**Comment by aismann** [comment]
Who want make a Wiki section about WASM?

**Reply by boris-741** [reply]
> How want make a Wiki section about WASM?

you're right. I didn't find anything in the docs about the run command
